tests/src/Context/ContentTypeContext.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\nexteuropa\Context\FieldContext.
+ */
+
+namespace Drupal\nexteuropa\Context;
+
+use Behat\Behat\Context\Context;
+use Behat\Gherkin\Node\TableNode;
+
+/**
+ * Context with content type management.
+ */
+class ContentTypeContext implements Context {
+
+  /**
+   * The list of node type before a scenario starts.
+   *
+   * @var array
+   */
+  protected $defaultNodeTypes = array();
+
+  /**
+   * Remember the list of node type.
+   *
+   * @BeforeScenario
+   */
+  public function rememberCurrentNodeTypes() {
+    foreach (node_type_get_types() as $result) {
+      $this->defaultNodeTypes[$result->type] = $result->type;
+    }
+  }
+
+  /**
+   * Removes any node types created after the last list node type remembered.
+   *
+   * @AfterScenario
+   */
+  public function resetNodeTypes() {
+    if (empty($this->defaultNodeTypes)) {
+      return;
+    }
+
+    foreach (node_type_get_types() as $result) {
+      if (!in_array($result->type, $this->defaultNodeTypes)) {
+        node_type_delete($result->type);
+      }
+    }
+    $this->defaultNodeTypes = array();
+  }
+
+  /**
+   * Add a field to a specified content type.
+   *
+   * @param string $arg1
+   *    The machine name of the content type to which attaching the field.
+   * @param TableNode $settings
+   *    Two columns table containing field settings.
+   *
+   * @Given a field with the following settings is added to the :arg1 type:
+   *
+   * @throws \InvalidArgumentException
+   *    If the field exist already in the Drupal system.
+   */
+  public function aFieldWithTheFollowingSettingsIsAddedToTheType($arg1, TableNode $settings) {
+    $field_info = array();
+    $field_instance_info = array();
+
+    $field_instance_info['entity_type'] = 'node';
+    $field_instance_info['bundle'] = $arg1;
+
+    // Assign fields to user before creation.
+    foreach ($settings->getRowsHash() as $setting_name => $value) {
+      switch ($setting_name) {
+        case 'Label':
+          $field_instance_info['label'] = $value;
+          break;
+
+        case 'Name':
+          $field_name = 'field_' . $value;
+          $is_field_exiting = field_info_field($field_name);
+          if ($is_field_exiting) {
```

Comment:
My previous comment on this kind of remarks is hidden because of new commits.

I do not use assert here because this step is just a preparatory step for the actual test comping just after. 
In this case, this test is justified for simplifying the definition of the test during the development phase; it is not intended to be a validation step.
Then, the use of an exception is more appropriate.